src/main/java/com/avispl/symphony/dal/infrastructure/management/tripleplay/TriplePlayAggregatorCommunicator.java

```diff
@@ -0,0 +1,601 @@
+	 */
+	private int calculatingLocalPollingInterval() {
+
+		try {
+			int pollingIntervalValue = TriplePlayConstrant.MIN_POLLING_INTERVAL;
+			if (StringUtils.isNotNullOrEmpty(pollingInterval)) {
+				pollingIntervalValue = Integer.parseInt(pollingInterval);
+			}
+
+			int minPollingInterval = calculatingMinPollingInterval();
+			if (pollingIntervalValue < minPollingInterval) {
+				logger.error(String.format("invalid pollingInterval value, pollingInterval must greater than: %s", minPollingInterval));
+				return minPollingInterval;
+			}
+			return pollingIntervalValue;
+		} catch (Exception e) {
+			throw new IllegalArgumentException(String.format("Unexpected pollingInterval value: %s", pollingInterval));
+		}
+	}
+
+	/**
+	 * Retrieve information of all clients
+	 *
+	 * @param currentSizeCachedClients Current size of cachedClients List, this param use to check number of clients was change or not
+	 */
+	private void retrieveInformationOfAllClients(int currentSizeCachedClients) {
+		if (executorService == null || currentSizeCachedClients != cachedClients.size()) {
+			executorService = Executors.newFixedThreadPool(deviceStatisticsCollectionThreads);
+		}
+
+		String lastClientKey = lastClientMac != null ? cachedClients.floorKey(lastClientMac) : null;
+		if (cachedClients.size() == 0) {
+			lastClientMac = null;
+			return;
+		}
+
+		Iterator<Entry<String, Client>> cacheClientIt = cachedClients.entrySet().iterator();
+
+		String itClient = null;
+		if (lastClientKey != null) {
+			while (cacheClientIt.hasNext()) {
+				if (cacheClientIt.next().getKey() == lastClientKey) {
+					break;
+				}
+			}
+		}
+
+		for (int threadNumber = 0; threadNumber < deviceStatisticsCollectionThreads && cacheClientIt.hasNext(); ++threadNumber) {
+			List<String> requestClientMac = new ArrayList<>();
+			for (int clientNumber = 0; clientNumber < TriplePlayConstrant.MAX_CLIENT_QUANTITY_PER_THREAD && cacheClientIt.hasNext(); ++clientNumber) {
+				itClient = cacheClientIt.next().getKey();
+				requestClientMac.add(itClient);
+			}
+			executorService.submit(new ClientLoader(requestClientMac));
+		}
+
+		//Update next ClientIp to next getMultipleStatistics
+		lastClientMac = itClient != cachedClients.lastKey() ? itClient : null;
+	}
+
+	/**
+	 * Retrieve all service
+	 */
+	private void retrieveService() {
+		try {
+			GetAllServicesRequestBody requestBody = new GetAllServicesRequestBody();
+			requestBody.setJsonrpc(TriplePlayConstrant.JSON_RPC);
+			requestBody.setParam(-1);
+			String response = doPost(TriplePlayURL.BASE_URI, requestBody.buildRequestBody());
+			ServiceWrapper monitoringData = objectMapper.readValue(response, ServiceWrapper.class);
+			for (Service service : monitoringData.getServices()) {
+				cachedServices.put(String.valueOf(service.getId()), service);
+			}
+			logger.debug(cachedServices);
+		} catch (Exception e) {
+			throw new ResourceNotReachableException(e.getMessage(), e);
```

Comment:
provide the meaning error message, such as : 
String.format("Error while retrieving the services from device: %s", e.getMessage)

Reply:
I fixed it